integration/query_fuzz_test.go

```diff
@@ -1354,6 +1355,30 @@ func runQueryFuzzTestCases(t *testing.T, ps *promqlsmith.PromQLSmith, c1, c2 *e2
 				break
 			}
 		}
+
+		nonDeterministicQuery := false
+		for i := 0; i < runForNonDeterCheck; i++ {
+			r1, err1 := c2.Query(query, queryTime)
+			r2, err2 := c2.Query(query, queryTime)
```

Comment:
Is it sufficient to compare the length of the series?

Why do `bottom` or `topk` query results differ? I was thinking the reason is non-determinism. But, I feel like my thinking might be wrong.

Reply:
As you said, it is non deterministic but the non deterministic is not highly I think. If you run 100 queries probably happen once or twice. I think running the same query with c2 is also highly likely to produce the same result so this can be still flaky.

> Is it sufficient to compare the length of the series?

We can add other verification if find doable but also reduce flakiness. At least comparing series length is better than not testing those functions at all.

Reply:
Ok, yes.. the same query with c2 is also flaky.
What about comparing # of samples? The series length is also flaky.

Reply:
I don't think number of samples is guaranteed the same. We randomized the timestamp of the first sample from each series so depending on your query you might be able to query some series but miss other series.

Reply:
Can we change promsmith to not use bottomk/topk with some "time based" functions (`day_of_week`, `time_of_day`, etc) ? i think this is the bulk of the errors?

Reply:
@alanprot 
I'm suspicious of all the same values. I tested the query `bottomk by (job) (pi(), day_of_year(-{__name__="test_series_a"} offset 1m17s))` where I changed the values to return a random value (1~366) of `day_of_year`.

Then, it is fine.
Btw, it sounds good to create the fuzz test testing those functions.